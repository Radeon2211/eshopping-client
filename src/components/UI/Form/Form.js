import React, { useCallback } from 'react';
import PropTypes from 'prop-types';
import { Form as FormikForm } from 'formik';
import { useSelector, useDispatch } from 'react-redux';
import * as actions from '../../../store/actions/indexActions';
import * as SC from './Form.sc';
import Button from '../Button/Button';
import Heading from '../Heading/Heading';

const Form = (props) => {
  const { headingText, btnText, isValid, cancellable, children, height } = props;

  const isFormLoading = useSelector((state) => state.ui.isFormLoading);
  const formError = useSelector((state) => state.ui.formError);

  const dispatch = useDispatch();
  const onSetModal = useCallback(
    (isModalOpen, modalContent) => dispatch(actions.setModal(isModalOpen, modalContent)),
    [dispatch],
  );

  const error = formError ? <span className="error">{formError}</span> : null;

  let heading = null;
  if (headingText) {
    heading = <Heading variant="h3">{headingText}</Heading>;
  }

  let cancelButton = null;
  if (cancellable) {
    cancelButton = (
      <div className="cancel-button-box">
        <Button clicked={() => onSetModal(false, '')}>cancel</Button>
      </div>
    );
  }

  let buttonsBox = null;
  if (btnText) {
    buttonsBox = (
      <div className="buttons-box">
        {cancelButton}
        <Button filled type="submit" disabled={!isValid || isFormLoading}>
          {btnText}
        </Button>
      </div>
    );
  }

  const form = (
    <FormikForm className="formik-form">
      {children}
      {buttonsBox}
    </FormikForm>
  );

  return (
    <SC.Wrapper height={height}>
      {heading}
      {form}
      {error}
    </SC.Wrapper>
  );
};

Form.defaultProps = {
  headingText: '',
  btnText: '',
  isValid: undefined,
  cancellable: false,
  height: undefined,
};

Form.propTypes = {
  headingText: PropTypes.string,
  btnText: PropTypes.string,
  isValid: PropTypes.bool,
  cancellable: PropTypes.bool,
  children: PropTypes.node.isRequired,
  height: PropTypes.number,
};

export default Form;
