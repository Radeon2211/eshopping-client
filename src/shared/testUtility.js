// eslint-disable-next-line import/no-extraneous-dependencies
import checkPropTypes from 'check-prop-types';

export const checkProps = (component, expectedProps) => {
  // eslint-disable-next-line react/forbid-foreign-prop-types
  return checkPropTypes(component.propTypes, expectedProps, 'props', component.name);
};

export const historyPageNum = (pageNumber) => ({
  listen: jest.fn(),
  location: { search: `?p=${pageNumber}` },
  createHref: jest.fn(),
  push: jest.fn(),
});

export const propsPagination = (itemQuantity = 5) => ({
  itemQuantity,
  isDataLoading: false,
  maxQuantityPerPage: 2,
});
