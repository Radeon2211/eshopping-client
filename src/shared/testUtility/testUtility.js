// eslint-disable-next-line import/no-extraneous-dependencies
import checkPropTypes from 'check-prop-types';
import { v4 as uuidv4 } from 'uuid';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from '../../store/reducers/rootReducer';
import { productConditions } from '../constants';

export const defaultDeliveryAddress = {
  firstName: 'firstName',
  lastName: 'lastName',
  street: 'street',
  zipCode: 'zipCode',
  city: 'city',
  country: 'country',
  phone: 'phone',
};

export const defaultUserProfile = {
  ...defaultDeliveryAddress,
  _id: 'userId',
  username: 'username',
  email: 'email@domain.com',
  status: 'active',
  contacts: {
    email: true,
    phone: true,
  },
  createdAt: '2021-02-10T19:10:38.872Z',
};

export const checkProps = (component, expectedProps) => {
  // eslint-disable-next-line react/forbid-foreign-prop-types
  return checkPropTypes(component.propTypes, expectedProps, 'props', component.name);
};

export const createHistoryPageNumber = (pageNumber = 1, push = jest.fn()) => ({
  listen: jest.fn(),
  location: { pathname: '/products', search: `?p=${pageNumber}` },
  createHref: jest.fn(),
  push,
});

export const createPaginationProps = (itemQuantity = 5) => ({
  itemQuantity,
  isDataLoading: false,
  quantityPerPage: 2,
});

export const createProductItem = (
  productId,
  sellerUsername = 'username',
  quantity = 1,
  price = 2,
  name = 'product name',
  photo = false,
  quantitySold = 0,
  buyerQuantity = 0,
  description = '',
  condition = productConditions.NEW,
) => ({
  _id: productId || uuidv4(),
  name,
  price,
  quantity,
  quantitySold,
  buyerQuantity,
  description,
  condition,
  photo,
  seller: {
    username: sellerUsername,
  },
  __v: 0,
  createdAt: '2021-02-10T19:10:38.872Z',
  updatedAt: '2021-02-10T19:10:38.872Z',
});

export const createCartItem = (
  sellerUsername = 'username',
  quantity = 1,
  productId,
  price = 2,
  name = 'product name',
  productQuantity = 2,
  photo = false,
) => {
  const product = createProductItem(
    productId || uuidv4(),
    sellerUsername,
    productQuantity,
    price,
    name,
    photo,
  );

  return {
    _id: uuidv4(),
    quantity,
    product,
  };
};

export const createTransactionAndOrderProdItem = (
  productId,
  sellerUsername = 'username',
  quantity = 1,
  price = 2,
  name = 'product name',
  photo = false,
) => ({
  _id: productId || uuidv4(),
  name,
  photo,
  price,
  quantity,
  seller: {
    username: sellerUsername,
  },
});

export const createOrder = (
  products = [createTransactionAndOrderProdItem()],
  orderId = uuidv4(),
  sellerUsername,
  buyerUsername,
  overallPrice,
  createdAt,
  sellerEmail,
  sellerPhone,
) => {
  const seller = sellerUsername
    ? {
        username: sellerUsername,
        email: sellerEmail,
        phone: sellerPhone,
      }
    : null;

  const buyer = buyerUsername
    ? {
        username: buyerUsername,
      }
    : null;

  return {
    _id: orderId,
    seller,
    buyer,
    products,
    overallPrice,
    deliveryAddress: defaultDeliveryAddress,
    createdAt,
    updatedAt: createdAt,
    __v: 0,
  };
};

export const testStore = (initialAuth = {}, initialProduct = {}, initialUI = {}) => {
  const createStoreWithMiddleware = applyMiddleware(thunk)(createStore);
  const defaultState = createStoreWithMiddleware(rootReducer).getState();

  const finalState = {
    auth: {
      ...defaultState.auth,
      ...initialAuth,
    },
    product: {
      ...defaultState.product,
      ...initialProduct,
    },
    ui: {
      ...defaultState.ui,
      ...initialUI,
    },
  };
  const finalStore = createStoreWithMiddleware(rootReducer, finalState);

  return {
    store: finalStore,
    initialState: finalStore.getState(),
  };
};

export const checkReqMethodAndURL = (moxiosInstance, method, url, reqNumber = 0) => {
  if (reqNumber) {
    const requests = moxiosInstance.requests.__items;
    const prevRequest = requests[reqNumber - 1];
    return method === prevRequest.config.method && url === prevRequest.config.url;
  }
  const lastRequest = moxiosInstance.requests.mostRecent();
  return method === lastRequest.config.method && url === lastRequest.config.url;
};

export const createExpectedState = (
  initialState,
  expectedAuth = {},
  expectedProduct = {},
  expectedUI = {},
) => {
  const newState = {
    auth: {
      ...initialState.auth,
      ...expectedAuth,
    },
    product: {
      ...initialState.product,
      ...expectedProduct,
    },
    ui: {
      ...initialState.ui,
      ...expectedUI,
    },
  };

  return newState;
};

export const setUpStoreWithDefaultProfile = (auth = {}, product = {}, ui = {}) => {
  return testStore(
    {
      profile: defaultUserProfile,
      deliveryAddress: defaultDeliveryAddress,
      cart: [],
      ...auth,
    },
    product,
    ui,
  );
};
