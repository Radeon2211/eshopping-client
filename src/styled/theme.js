const theme = {
  colors: {
    light1: '#FAFAFA',
    light1Transparent: '#FAFAFA88',
    light2: '#DADFE1',
    light3: '#BFBFBF',
    light4: '#666',
    blue: '#1E8BC3',
    blueLight: '#22A7F0',
    blueDark: '#67809F',
    green: '#00A790',
    greenDark: '#009680',
    red: '#F22613',
  },
  shadows: {
    level1: '0 0 2px rgba(0, 0, 0, .1)',
    level2: '0 0 2px rgba(0, 0, 0, .3)',
    level3: '0 0 3px rgba(0, 0, 0, .3)',
    level4: '0 0 5px rgba(0, 0, 0, .3)',
    level5: '0 0 8px rgba(0, 0, 0, .3)',
  },
  fonts: {
    logo: 'Lobster, cursive',
    heading: 'Ubuntu, sans-serif',
    text: 'Lato, sans-serif',
  },
  spacings: {
    level1: '0.8rem',
    level2: '1.2rem',
    level3: '2.4rem',
    level4: '3.6rem',
    level5: '4.8rem',
  },
  zIndexes: {
    level1: 10,
    level2: 100,
    level3: 1000,
    level4: 10000,
  },
};

export default theme;
