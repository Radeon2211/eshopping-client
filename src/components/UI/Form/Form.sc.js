import styled from 'styled-components';

// eslint-disable-next-line import/prefer-default-export
export const Wrapper = styled.div`
  align-self: flex-start;
  justify-self: center;
  max-width: 100%;
  text-align: left;
  width: 65rem;

  ${({ height }) => {
    if (height) {
      return `
        & .formik-form {
          position: relative;
          height: ${height}rem;
        }
      `;
    }
    return ``;
  }}
`;
