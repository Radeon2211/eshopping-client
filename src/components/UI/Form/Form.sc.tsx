import styled from 'styled-components';
import { FormProps } from './Form';

export const Wrapper = styled.div<{ $height: FormProps['height'] }>`
  align-self: flex-start;
  justify-self: center;
  max-width: 100%;
  text-align: left;
  width: 65rem;

  ${({ $height }) => {
    if ($height) {
      return `
        & .formik-form {
          position: relative;
          height: ${$height}rem;
        }
      `;
    }
    return ``;
  }}
`;
