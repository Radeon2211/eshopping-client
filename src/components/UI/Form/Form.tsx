import { useCallback } from 'react';
import { Form as FormikForm } from 'formik';
import * as actions from '../../../store/actions/indexActions';
import * as SC from './Form.sc';
import Button from '../Button/Button';
import FlexWrapper from '../FlexWrapper';
import PlainText from '../PlainText';
import Heading from '../Heading/Heading';
import theme from '../../../styled/theme';
import { ModalType, PropsWithChildren } from '../../../shared/types/types';
import { useTypedDispatch, useTypedSelector } from '../../../store/reducers/rootReducer';

export interface FormProps {
  headingText?: string;
  btnText?: string;
  isValid?: boolean;
  cancellable?: boolean;
  height?: number;
  btnColor?: 'blue' | 'red';
}

export default function Form(props: PropsWithChildren<FormProps>) {
  const {
    headingText,
    btnText,
    isValid,
    cancellable = false,
    children,
    height,
    btnColor = 'blue',
  } = props;

  const isFormLoading = useTypedSelector((state) => state.ui.isFormLoading);
  const formError = useTypedSelector((state) => state.ui.formError);

  const dispatch = useTypedDispatch();
  const onSetModal = useCallback(
    (modalContent: ModalType | null) => dispatch(actions.setModal(modalContent)),
    [dispatch],
  );

  const error = formError ? (
    <PlainText $size="level3" $mgTop="level3" $color={theme.colors.red} data-testid="Form-error">
      {formError}
    </PlainText>
  ) : null;

  let heading = null;
  if (headingText) {
    heading = <Heading $variant="h3">{headingText}</Heading>;
  }

  let cancelButton = null;
  if (cancellable) {
    cancelButton = (
      <PlainText $display="block" $mgRight="level3">
        <Button clicked={() => onSetModal(null)}>cancel</Button>
      </PlainText>
    );
  }

  let buttonsBox = null;
  if (btnText) {
    buttonsBox = (
      <FlexWrapper $align="center" $justify="flex-end" $spacing="level3">
        {cancelButton}
        <Button
          $filled
          type="submit"
          $color={btnColor}
          disabled={!isValid || isFormLoading}
          data-testid="Form-submit-btn"
        >
          {btnText}
        </Button>
      </FlexWrapper>
    );
  }

  const form = (
    <FormikForm className="formik-form">
      {children}
      {buttonsBox}
    </FormikForm>
  );

  return (
    <SC.Wrapper $height={height}>
      {heading}
      {form}
      {error}
    </SC.Wrapper>
  );
}
