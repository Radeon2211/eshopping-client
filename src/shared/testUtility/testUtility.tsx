/* eslint-disable import/no-extraneous-dependencies */
import { ReactElement } from 'react';
import { Router } from 'react-router-dom';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { ThemeProvider } from 'styled-components';
import { Provider } from 'react-redux';
import { MockStoreEnhanced } from 'redux-mock-store';
import { fireEvent, waitFor, render } from '@testing-library/react';
import { v4 as uuidv4 } from 'uuid';
import moxios from 'moxios';
import rootReducer from '../../store/reducers/rootReducer';
import theme from '../../styled/theme';
import {
  CartItem,
  DeliveryAddress,
  Order,
  OrderProductItem,
  Product,
  Profile,
  ProfileStatus,
} from '../types/types';
import { ProductCondition } from '../types/enums';

export const defaultDeliveryAddress: DeliveryAddress = {
  firstName: 'firstName',
  lastName: 'lastName',
  street: 'street',
  zipCode: 'zipCode',
  city: 'city',
  country: 'country',
  phone: 'phone',
};

export const defaultUserProfile: Profile = {
  ...defaultDeliveryAddress,
  _id: 'userId',
  username: 'username',
  email: 'email@domain.com',
  status: ProfileStatus.ACTIVE,
  contacts: {
    email: true,
    phone: true,
  },
  createdAt: '2021-02-10T19:10:38.872Z',
};

export const createPaginationProps = (itemQuantity = 5) => ({
  itemQuantity,
  isDataLoading: false,
  quantityPerPage: 2,
});

export const createProductItem = (props: Partial<Product> = {}): Product => {
  return {
    _id: uuidv4(),
    quantity: 1,
    price: 2,
    name: 'product name',
    photo: false,
    quantitySold: 0,
    buyerQuantity: 0,
    description: '',
    condition: ProductCondition.NEW,
    seller: {
      username: 'username',
    },
    createdAt: '2021-02-10T19:10:38.872Z',
    updatedAt: '2021-02-10T19:10:38.872Z',
    __v: 0,
    ...props,
  };
};

export const createCartItem = (props = {}): CartItem => {
  const finalProps = {
    id: uuidv4(),
    sellerUsername: 'username',
    quantity: 1,
    productId: uuidv4(),
    price: 2,
    name: 'product name',
    productQuantity: 2,
    photo: false,
    ...props,
  };
  const product = createProductItem({
    _id: finalProps.productId,
    quantity: finalProps.productQuantity,
    price: finalProps.price,
    name: finalProps.name,
    photo: finalProps.photo,
    seller: {
      username: finalProps.sellerUsername,
    },
  });

  return {
    _id: finalProps.id,
    quantity: finalProps.quantity,
    product,
  };
};

export const createTransactionAndOrderProdItem = (
  props: Partial<OrderProductItem> = {},
): OrderProductItem => {
  return {
    _id: uuidv4(),
    quantity: 1,
    price: 2,
    name: 'product name',
    photo: false,
    seller: {
      username: 'username',
    },
    ...props,
  };
};

export const createOrder = (props: Partial<Order> = {}): Order => {
  return {
    _id: uuidv4(),
    overallPrice: 100,
    seller: null,
    buyer: null,
    products: [createTransactionAndOrderProdItem()],
    deliveryAddress: defaultDeliveryAddress,
    createdAt: '2021-02-10T19:10:38.872Z',
    updatedAt: '2021-02-10T19:10:38.872Z',
    __v: 0,
    ...props,
  };
};

export const testStore = (initialAuth = {}, initialProduct = {}, initialUI = {}) => {
  const createStoreWithMiddleware = applyMiddleware(thunk)(createStore);
  const defaultState = createStoreWithMiddleware(rootReducer).getState();

  const finalState = {
    auth: {
      ...defaultState.auth,
      ...initialAuth,
    },
    product: {
      ...(defaultState.product as object),
      ...initialProduct,
    },
    ui: {
      ...(defaultState.ui as object),
      ...initialUI,
    },
  };
  // @ts-expect-error TBF after redux upgrade
  const finalStore = createStoreWithMiddleware(rootReducer, finalState);

  return {
    store: finalStore,
    initialState: finalStore.getState(),
  };
};

type MoxiosRequestsItems = { config: { method: string; url: string } }[];
type MockedMoxiosInstance = {
  requests: {
    __items?: MoxiosRequestsItems;
    mostRecent?: () => { config: { method: string; url: string } };
  };
};
type DefaultMoxiosInstance = Omit<typeof moxios, 'requests'> & {
  requests: typeof moxios.requests & {
    __items?: MoxiosRequestsItems;
  };
};
export const checkReqMethodAndURL = (
  moxiosInstance: MockedMoxiosInstance | DefaultMoxiosInstance,
  method: string,
  url: string,
  reqNumber = 0,
) => {
  if (reqNumber) {
    const requests = moxiosInstance.requests.__items;
    const prevRequest = requests?.[reqNumber - 1];
    return prevRequest && method === prevRequest.config.method && url === prevRequest.config.url;
  }
  if (!moxiosInstance.requests.mostRecent) return false;
  const lastRequest = moxiosInstance.requests.mostRecent();
  return method === lastRequest.config.method && url === lastRequest.config.url;
};

export const createExpectedState = (
  initialState: {
    auth?: object;
    product?: object;
    ui?: object;
  },
  expectedAuth = {},
  expectedProduct = {},
  expectedUI = {},
) => {
  const newState = {
    auth: {
      ...initialState.auth,
      ...expectedAuth,
    },
    product: {
      ...initialState.product,
      ...expectedProduct,
    },
    ui: {
      ...initialState.ui,
      ...expectedUI,
    },
  };

  return newState;
};

export const setUpStoreWithDefaultProfile = (auth = {}, product = {}, ui = {}) => {
  return testStore(
    {
      profile: defaultUserProfile,
      deliveryAddress: defaultDeliveryAddress,
      cart: [],
      ...auth,
    },
    product,
    ui,
  );
};

export const clickAtSubmitButton = async (container: Element) => {
  await waitFor(() => {
    const submitBtn = container.querySelector('button[type="submit"]');
    if (submitBtn) {
      fireEvent.click(submitBtn);
    }
  });
};

export const testRouterPushCall = (
  pushFn: ReturnType<typeof jest.fn>,
  callIdx: number,
  expectedPathname: string,
  expectedSearch = '',
) => {
  expect(pushFn.mock.calls[callIdx][0].pathname).toEqual(expectedPathname);
  expect(pushFn.mock.calls[callIdx][0].search).toEqual(expectedSearch);
};

type RenderAppPartOptions = {
  push?: ReturnType<typeof jest.fn>;
  pathname?: string;
  search?: string;
  withoutThemeProvider?: boolean;
  withoutRouter?: boolean;
  store?: MockStoreEnhanced<object, object>;
  onlyReturnWrappedElement?: boolean;
};

export const renderAppPart = (element: ReactElement, options: RenderAppPartOptions = {}) => {
  const navigator = {
    createHref: jest.fn(),
    push: options.push || jest.fn(),
    go: jest.fn(),
    replace: jest.fn(),
  };

  const location = {
    pathname: options.pathname || '/',
    search: options.search || '',
  };

  let content = element;
  if (!options.withoutThemeProvider) {
    content = <ThemeProvider theme={theme}>{element}</ThemeProvider>;
  }
  if (options.store) {
    content = <Provider store={options.store}>{content}</Provider>;
  }
  if (!options.withoutRouter) {
    content = (
      <Router location={location} navigator={navigator}>
        {content}
      </Router>
    );
  }

  return options.onlyReturnWrappedElement ? content : render(content);
};
