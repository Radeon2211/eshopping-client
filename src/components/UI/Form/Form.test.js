import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { ThemeProvider } from 'styled-components';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import { Formik } from 'formik';
import theme from '../../../styled/theme';
import Form from './Form';
import { checkProps } from '../../../shared/testUtility';

const mockStore = configureMockStore([thunk]);

const defaultProps = {
  headingText: 'testHeadingText',
  btnText: 'submitText',
  isValid: true,
  cancellable: true,
  children: <div data-testid="Form-children" />,
};

const createStore = (isFormLoading, formError) =>
  mockStore({
    ui: {
      isFormLoading,
      formError,
    },
  });

const defaultStore = createStore(false, 'testError');

const setUp = (props, store) => {
  return render(
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <Formik>
          <Form {...props} />
        </Formik>
      </ThemeProvider>
    </Provider>,
  );
};

afterEach(cleanup);

describe('<Form />', () => {
  describe('Check prop types', () => {
    it('Should NOT throw a warning', () => {
      const expectedProps = {
        children: <div />,
      };
      expect(checkProps(Form, expectedProps)).toBeUndefined();
    });

    it('Should throw a warning', () => {
      const expectedProps = {};
      expect(checkProps(Form, expectedProps)).not.toBeNull();
    });
  });

  describe('Check how renders', () => {
    it('Should render everything what is possible with blue submit button', () => {
      const { asFragment } = setUp(defaultProps, defaultStore);
      expect(asFragment()).toMatchSnapshot();
    });

    it('Should NOT render heading, error and buttons', () => {
      const props = {
        ...defaultProps,
        headingText: '',
        btnText: '',
        cancellable: false,
      };
      const store = createStore(false, '');
      const { asFragment } = setUp(props, store);
      expect(asFragment()).toMatchSnapshot();
    });

    it('Should render red submit button without cancel button', () => {
      const props = {
        ...defaultProps,
        headingText: '',
        cancellable: false,
        btnColor: 'red',
      };
      const store = createStore(false, '');
      const { asFragment } = setUp(props, store);
      expect(asFragment()).toMatchSnapshot();
    });

    it('Should submit button be disabled if isFormLoading is true', () => {
      const store = createStore(true, '');
      setUp(defaultProps, store);
      expect(screen.getByTestId('Form-submit-btn')).toBeDisabled();
    });

    it('Should submit button be disabled if isValid is false', () => {
      const props = {
        ...defaultProps,
        isValid: false,
      };
      setUp(props, defaultStore);
      expect(screen.getByTestId('Form-submit-btn')).toBeDisabled();
    });

    it('Should submit button NOT be disabled if isValid is true and isFormLoading is false', () => {
      setUp(defaultProps, defaultStore);
      expect(screen.getByTestId('Form-submit-btn')).not.toBeDisabled();
    });
  });
});
