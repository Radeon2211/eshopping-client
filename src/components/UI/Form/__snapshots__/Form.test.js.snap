// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> Check how renders Should NOT render heading, error and buttons 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm fKgwDr"
  >
    <form
      action="#"
      class="formik-form"
    >
      <div
        data-testid="Form-children"
      />
    </form>
  </div>
</DocumentFragment>
`;

exports[`<Form /> Check how renders Should render everything what is possible with blue submit button 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm fKgwDr"
  >
    <h3
      class="sc-Axmtr kmnqOV"
    >
      testHeadingText
    </h3>
    <form
      action="#"
      class="formik-form"
    >
      <div
        data-testid="Form-children"
      />
      <div
        class="sc-AxheI drMepa"
        direction="row"
        spacing="3"
        width="100%"
        wrap="nowrap"
      >
        <span
          class="sc-AxhUy kBnmbO"
          color=""
          display="block"
          overflow="visible"
          spacing="0px"
          transform=""
        >
          <button
            class="sc-AxirZ MoZHP"
            color="blue"
            type="button"
          >
            cancel
          </button>
        </span>
        <button
          class="sc-AxirZ BxhX"
          color="blue"
          data-testid="Form-submit-btn"
          type="submit"
        >
          submitText
        </button>
      </div>
    </form>
    <span
      class="sc-AxhUy gXOcQt"
      color="#F22613"
      display="inline-block"
      overflow="visible"
      size="3"
      spacing="0px"
      transform=""
    >
      testError
    </span>
  </div>
</DocumentFragment>
`;

exports[`<Form /> Check how renders Should render red submit button without cancel button 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm fKgwDr"
  >
    <form
      action="#"
      class="formik-form"
    >
      <div
        data-testid="Form-children"
      />
      <div
        class="sc-AxheI drMepa"
        direction="row"
        spacing="3"
        width="100%"
        wrap="nowrap"
      >
        <button
          class="sc-AxirZ dFUlZb"
          color="red"
          data-testid="Form-submit-btn"
          type="submit"
        >
          submitText
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
